beam/task/backend/test/Test.hs

```diff
@@ -21,14 +23,26 @@ import Database.PostgreSQL.Serializable
 import Gargoyle.PostgreSQL.Connect
 import System.FilePath
 import System.Posix.Temp
+import System.Timeout
 import Test.Hspec
+import Test.HUnit.Lang
 
 import Rhyolite.Task.Beam
 import Rhyolite.Task.Beam.Worker
 
 import Types
 import Utils
 
+seconds :: Num a => a -> a
+seconds = (* 1e6)
+
+threadDelayPolling :: Int -> Int -> Expectation -> Expectation
+threadDelayPolling delayTimeout pollInterval test = void $ timeout delayTimeout go
+  where
+    go = do
+      threadDelay pollInterval
+      catch test $ \(e :: HUnitFailure) -> go
```

Comment:
Note this is stated to be an internal API:
 https://hackage.haskell.org/package/HUnit-1.6.2.0/docs/Test-HUnit-Lang.html#g:1
I didn't find a suitable non-internal equivalent though